python/feathub/tests/feathub_it_test_base.py

```diff
@@ -47,6 +51,37 @@ def _merge_nested_dict(a, b) -> None:
             )
 
 
+# A wrapper class for Registry that verifies the to/from json methods for
+# every saved table descriptor.
+class RegistryWithJsonCheck(Registry):
+    def __init__(self, registry: Registry):
+        super().__init__("", {})
+        self.registry = registry
+
+    def build_features(
+        self, features_list: List[TableDescriptor], props: Optional[Dict] = None
+    ) -> List[TableDescriptor]:
+        features_list = [self._save_and_reload_through_json(x) for x in features_list]
+        return self.registry.build_features(features_list, props)
+
+    def register_features(
+        self, features: TableDescriptor, override: bool = True
+    ) -> bool:
+        features = self._save_and_reload_through_json(features)
+        return self.registry.register_features(features, override)
+
+    def get_features(self, name: str) -> TableDescriptor:
+        return self.registry.get_features(name)
+
+    def delete_features(self, name: str) -> bool:
+        return self.delete_features(name)
+
+    @staticmethod
+    def _save_and_reload_through_json(features: TableDescriptor):
+        json_bytes = json.dumps(features.to_json(), sort_keys=True).encode("utf8")
```

Comment:
Can you explain why we need to use `encode("utf8")`? The Python doc does not seem to use this.

[1] https://docs.python.org/3/library/json.html